Apps who depends on `swift-composable-architecture` build failed if `minimum_os_version >= 17.0`

I'm not sure whether it's a `rules_swift_package_manager` issue or `rules_swift` issue.

[`swift-composable-architecture`](https://github.com/pointfreeco/swift-composable-architecture) has a dependency [`swift-perception`](https://github.com/pointfreeco/swift-perception).
[`swift-perception`](https://github.com/pointfreeco/swift-perception) is a backport library for [`Observation`](https://developer.apple.com/documentation/observation). Since `Observation` requires iOS 17+, some API in `swift-perception` are marked obsoleted on iOS 17. eg. [`Bindable`](https://github.com/pointfreeco/swift-perception/blob/main/Sources/Perception/Bindable.swift#L8-L15)
```
  @available(iOS, introduced: 13, obsoleted: 17)
  @available(macOS, introduced: 10.15, obsoleted: 14)
  @available(tvOS, introduced: 13, obsoleted: 17)
  @available(watchOS, introduced: 6, obsoleted: 10)
  @available(visionOS, unavailable)
  @dynamicMemberLookup
  @propertyWrapper
  public struct Bindable<Value>
```

---

[the TCA Sample](https://github.com/cgrindel/rules_swift_package_manager/tree/main/examples/tca_example) build succeeds when [`minimum_os_version = "16.0"`](https://github.com/cgrindel/rules_swift_package_manager/blob/e93a1243ce31057bedb224e5d5da5748439b0225/examples/tca_example/Sources/BUILD.bazel#L22C5-L22C33).
However, if the line is changed to `minimum_os_version = "17.0"`, errors like the following will occur.
```
error: emit-module command failed with exit code 1 (use -v to see invocation)
external/rules_swift_package_manager~override~swift_deps~swiftpkg_swift_perception/Sources/Perception/Bindable.swift:21:32: error: 'Bindable' is unavailable in iOS
    public var projectedValue: Bindable<Value> {
                               ^~~~~~~~
external/rules_swift_package_manager~override~swift_deps~swiftpkg_swift_perception/Sources/Perception/Bindable.swift:15:17: note: 'Bindable' was obsoleted in iOS 17
  public struct Bindable<Value> {
                ^
external/rules_swift_package_manager~override~swift_deps~swiftpkg_swift_perception/Sources/Perception/Bindable.swift:36:63: warning: 'Perceptible' was deprecated in iOS 17: renamed to 'Observable'
    public init(wrappedValue: Value) where Value: AnyObject & Perceptible {
                                                              ^
external/rules_swift_package_manager~override~swift_deps~swiftpkg_swift_perception/Sources/Perception/Bindable.swift:36:63: note: use 'Observable' instead
    public init(wrappedValue: Value) where Value: AnyObject & Perceptible {
                                                              ^~~~~~~~~~~
                                                              Observable
```
While the minimum deployments versions of [some TCA examples](https://github.com/pointfreeco/swift-composable-architecture/tree/main/Examples) are also >= 17.0, and they can build successfully in Xcode & Swift Packages.

--- a/examples/tca_example/Sources/BUILD.bazel
+++ b/examples/tca_example/Sources/BUILD.bazel
@@ -18,7 +18,7 @@ ios_application(
         "ipad",
     ],
     infoplists = [":Info.plist"],
-    minimum_os_version = "16.0",
+    minimum_os_version = "17.0",
     visibility = ["//visibility:public"],
     deps = [":TCAExample"],
 )

--- a/examples/tca_example/Tests/BUILD.bazel
+++ b/examples/tca_example/Tests/BUILD.bazel
@@ -15,7 +15,7 @@ swift_library(
 
 ios_unit_test(
     name = "TCAExampleTests",
-    minimum_os_version = "16.0",
+    minimum_os_version = "17.0",
     runner = "@build_bazel_rules_apple//apple/testing/default_runner:ios_xctestrun_ordered_runner",
     visibility = ["//visibility:private"],
     deps = [":TCAExampleTestsLib"],

